libs/variant/include/variant/variant.hpp

```diff
@@ -596,9 +593,9 @@ struct ForwardSerializer<fetch::variant::Variant, D>
   template <typename Serializer>
   static void Serialize(Serializer &serializer, Type const &var)
   {
-    auto typecode = static_cast<int>(var.type());
+    auto typecode = static_cast<int>(var.type_);
```

Comment:
this forces friend class. The public interface method `type()` gives the same performance as in optimized release build as direct access.